compiler/rustc_lint/src/lints.rs

```diff
@@ -753,15 +753,37 @@ impl AddToDiagnostic for HiddenUnicodeCodepointsDiagSub {
 #[diag(lint_map_unit_fn)]
 #[note]
 pub struct MappingToUnit {
-    #[label(lint_function_label)]
-    pub function_label: Span,
     #[label(lint_argument_label)]
     pub argument_label: Span,
     #[label(lint_map_label)]
     pub map_label: Span,
     #[suggestion(style = "verbose", code = "{replace}", applicability = "maybe-incorrect")]
     pub suggestion: Span,
     pub replace: String,
+    #[subdiagnostic]
+    pub func_label: MappingToUnitFuncLabel,
+}
+
+pub struct MappingToUnitFuncLabel {
```

Comment:
Can you make an enum with two variants, and then you can `derive(Subdiagnostic)`